geopandas/geodataframe.py

```diff
@@ -587,7 +587,7 @@ def __getitem__(self, key):
         """
         result = super(GeoDataFrame, self).__getitem__(key)
         geo_col = self._geometry_column_name
-        if isinstance(key, str) and key == geo_col:
+        if isinstance(result, Series) and isinstance(result.dtype, GeometryDtype):
             result.__class__ = GeoSeries
             result.crs = self.crs
```

Comment:
As long as https://github.com/geopandas/geopandas/pull/1339 is not in, should we rather only set the CRS when `key == geo_col` ? 
Or do we now implicitly assume that the `crs` holds for all geometry columns?

Reply:
We currently do implicitly assume that `crs` holds for all. Doing `set_geometry` expects exactly that by default. I think it is a better solution than returning GeoSeries without CRS. I assume #1339 will be part of next release, so it does not matter much anyway.